.github/workflows/test-deploy.yml

```diff
@@ -0,0 +1,34 @@
+name: Test & Deploy
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.4, 3.5, 3.6, 3.7, 3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install poetry
+        run: |
+            curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python
+            source $HOME/.poetry/env
+      - name: Install dependencies
+        run: poetry install
+      - name: Run Tox
+        run: poetry run tox -e py  # Run tox using the version of Python in `PATH`
+      - name: Build package
+        run: poetry build
+      - name: Deploy to PyPI
+        run: poetry publish
+        continue-on-error: true
+        if: success() && github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
This will currently only publish new versions when the version provided in pyproject.toml, is there any case where you'd modify the package version but will not want to update the package?

Reply:
I guess it depends on the individual workflow. I like git tags, because it's a very easy way to annotate individual commits with versions. And then I would want the pipeline to follow that. Also, I don't think running a command called "publish" for every pipeline is the right semantics. Then it should be called "try_publish" or "publish_if_new" or something like that. But maybe that's just me.

Reply:
So just to make sure I understand: If I now run poetry publish and this version already exists, it will not do anything? If yes, let's merge this as it is, and I'll figure out how to change the pipeline myself, if I feel like it. I don't want to stand in the way of a CI, which I think is awesome!

Reply:
So what happens is that it will upload the packages but pypi will just reject it, essentially, because you can't update a release, understandably. So I will merge this now. But I still want to change this, eventually. But it has slipped down on my list of priorities.

Reply:
Yes, that's exactly how it works.

I've looked into changing the system to only publish when new tags are created and the closest thing I've found is [this](https://github.community/t5/GitHub-Actions/How-to-run-GitHub-Actions-Workflow-only-for-new-tags/td-p/29413). The main problem is that if we were to use that approach we would need to have two actions that would essentially share pretty much almost the same code (install dependencies, build...), which would be kinda messy. There might be a better way to do it but I haven't been able to find it.

Reply:
Can you do something like `[ git describe HEAD = git tag ]`?

Reply:
> Can you do something like `[ git describe HEAD = git tag ]`?

Can I run arbitrary commands inside the `if` clause? From my understanding of [the docs](https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#about-context-and-expressions) this is not possible.

I might be wrong here tho, I don't use Gihub actions frequently and I'm not familiar with them. I've just been browsing the docs to build this action and I've probably missed some things.

Reply:
let's leave it. What do I need to do to make this repo actually execute these actions?

Reply:
Skimming through that it seems to me that you could probably work around such limitations easily enough, along these lines:
```yaml
  - name: is this a tag?
    run: test "$(git describe HEAD)" = "$(git tag)"
    if: success() && github.event_name == 'push' && github.ref == 'refs/heads/master'

  - name: Deploy to PyPI
    run: poetry publish
    continue-on-error: true
    if: success()
```

Reply:
PS I'm not objecting to anything here; just trying to be potentially helpful. ;-)

Reply:
@hellerbarde Go to Settings > Actions and "Enable local and third party Actions for this repository"

![image](https://user-images.githubusercontent.com/32309574/75100196-b03f8700-5605-11ea-9baf-e0797711a2dc.png)

Reply:
@Frenzie Great idea! Do you mind testing it and submitting a PR? Also you might want to add a "continue-on-error: true" to the extra step so the whole build is not marked as a failure if there are no new tags.

Reply:
I can probably take a look tomorrow.